cluster/saltbase/salt/kube-proxy/default

```diff
@@ -15,5 +15,13 @@
     {% set ips = salt['mine.get']('roles:kubernetes-master', 'network.ip_addrs', 'grain').values() -%}
     {% set api_servers = "--master=https://" + ips[0][0] -%}
   {% endif -%}
+
+  # TODO: remove nginx for other cloud providers.
+  {% if grains['cloud'] is defined and grains.cloud in [ 'aws', 'gce' ]  %}
+     {% set api_servers_with_port = api_servers -%}
+  {% else -%}
+    {% set api_servers_with_port = api_servers + ":6443" -%}
+  {% endif -%}
+
 {% endif -%}
-DAEMON_ARGS="{{daemon_args}} {{api_servers}} {{kubeconfig}} {{pillar['log_level']}}"
+DAEMON_ARGS="{{daemon_args}} {{api_servers_with_port}} {{kubeconfig}} {{pillar['log_level']}}"
```

Comment:
You need to change line 8 to set api_servers_with_port or this will break azure.